### PR TITLE
Expand & update documentation on upgrading JupyterHub

### PR DESCRIPTION
**Pull request by yuvipanda**
- Expands the previous documentation on upgrading JupyterHub
  to include more information.
- Remove specific documentation on 0.7 -> 0.8 upgrade, since
  this seems to be a straight copy of the markdown version of
  upgrading docs. The important thing about the 0.7 -> 0.8 upgrade
  (requiring versions of JupyterHub to match) is now in the
  main document.
- Move from markdown to rst
- Remove the 'tutorials' page, which is now empty
- Add more prominent link to 'distributions'

Ref #2160

### CODE REVIEW

File: docs/source/admin/upgrading.rst [hunk 1]
```diff
@@ -0,0 +1,159 @@
+.. _admin/upgrading:
+
+====================
+Upgrading JupyterHub
+====================
+
+JupyterHub offers easy upgrade pathways between minor versions. This
+document describes how to do these upgrades. 
+
+If you are using a JupyterHub distribution, you should consult the
```
[the remark this answers is not in the document]
Comment (betatim) [reply docs/source/admin/upgrading.rst]: maybe add a definition for what a JupyterHub distribution is:

"... are using a JupyterHub distribution like [The Littlest JupyterHub](linky link) or [Zero2JupyterHub](link me), you should consult ..."

File: docs/source/admin/upgrading.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+If you are using the default configuration where ``configurable-http-proxy``
+is managed by JupyterHub, your users will see service disruption during
+the upgrade process. You should notify them, and pick a time to do the
+upgrade where they will be least disrupted.
+
+If you are using a different proxy, or running ``configurable-http-proxy``
+independent of JupyterHub, your users will be able to continue using notebook
+servers they had already launched, but will not be able to launch new servers
+nor sign in. This gives you more freedom to do upgrades without affecting
+users *too much*.
```
Comment (betatim) [inline docs/source/admin/upgrading.rst]: I think the "too much" doesn't help the reader. What is "too much"?

Maybe replace the sentence with "During the upgrade users won't be able to login and start new session, existing notebook sessions will continue uninterrupted."?

File: docs/source/admin/upgrading.rst [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+   python3 -m pip install --upgrade jupyterhub==<version>
+
+Where ``<version>`` is the version of JupyterHub you are upgrading to.
+
+If you used ``conda`` to install ``jupyterhub``, you should upgrade it
+with:
+
+.. code-block:: bash
+
+   conda install jupyterhub==<version>
```
Comment (betatim) [inline docs/source/admin/upgrading.rst]: `conda install -c conda-forge jupyterhub==<version>` to match the README's recommendation

File: docs/source/admin/upgrading.rst (hunk 2 above)
Comment (betatim) [reply docs/source/admin/upgrading.rst]: Or link to the section on distributions in this documentation

**Comment by yuvipanda** [comment]
@betatim @choldgraf thanks for your comments! I've updated PR addressing them.

**Pull request merged**